src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -260,6 +272,21 @@ class ImplLookup(
             .map { SelectionCandidate.DerivedTrait(it) }
     }
 
+    private fun sizedTraitCandidates(ty: Ty, sizedTrait: RsTraitItem): Set<SelectionCandidate> {
+        var isSized = ty.isSized()
+        if (!isSized && ty is TyTypeParameter) {
+            isSized = ty.getTraitBoundsTransitively().any { it.element == sizedTrait }
```

Comment:
Hmm, why we need this branch?

Reply:
In this case
```rust
trait Foo : Sized {
    fn foo(self: Self);
}
```
`Self` type should be `Sized` and we need to check super traits.
But constructing of `TyTypeParameter.Self` is better place to do it so I've moved corresponding code there.

Reply:
Oh, Ok